papermage_components/table_structure_predictor_mathpix.py

```diff
@@ -0,0 +1,141 @@
+from typing import List
+import io
+import torch
+from torchvision import transforms
+from transformers import TableTransformerForObjectDetection
+
+from papermage import Box, Document, Entity, TablesFieldName
+from papermage.predictors import BasePredictor
+from papermage_components.utils import get_table_image, get_text_in_box, globalize_bbox_coordinates
+
+import requests
+import base64
+import json
+import csv
+from collections import defaultdict
+from PIL import Image
+
+headers = {
+  "app_id": "",
+  "app_key": ""
+}
+
+url = 'https://api.mathpix.com/v3/text'
+
+def get_mathpix_input(encoded_image):
+    json_data = {
+        "src": f"data:image/jpeg;base64,{encoded_image}",
+        "formats": ["text", "data"],
+        "data_options": {
+            "include_tsv": True,
+        }
+    }
+    return json_data
+
+
+def get_nearby_captions(table, doc, expansion_factor):
+    box = table.boxes[0]
+
+    exp_h = expansion_factor * box.h
+    diff_h = exp_h - box.h
+
+    search_box = Box(l=box.l, t=box.t - diff_h / 2, w=box.w, h=exp_h, page=box.page)
+    potential_captions = doc.find(query=search_box, name="captions")
+    return potential_captions
+
+
+class TableStructurePredictor(BasePredictor):
+    def get_table_test(self, doc: Document) -> List[Entity]:
+        img = []
+        for table in getattr(doc, TablesFieldName):
+            if abs(table.spans[0].start - table.spans[0].end)<100:
+                continue
+            table_image = get_table_image(table, doc)
+            img.append(table_image)
+        return img
+    def get_table(self, doc: Document) -> List[Entity]:
```

Comment:
For this to correctly implement the interface, it needs to be called `_predict()`, not `get_table()`

Reply:
switched from get_table() to _predict()